Animation beta

0b5vr opened the pull request:
This branch manages the beta release of three-vrm which contains VRM animation features.
It will be merged into `dev` once VRM animation spec is released.

0b5vr committed:
Revert "Revert "feature: VRMAnimation""

This reverts commit 2f09d712311f87281c183a44460f9db0a035d7e3.

0b5vr committed:
v2.1.0-beta.0

0b5vr committed:
fix: Fix the issue that hips translation goes infinity

I was probably quite sleepy when working on the restHipsPosition procedure last time

See: https://github.com/pixiv/three-vrm/commit/c11d6e9de74b1621292c5250cae7ae98cf26524a

0b5vr committed:
Merge pull request #1342 from pixiv/fix-animation-hips

fix: Fix the issue that hips translation goes infinity

0b5vr committed:
v2.1.0-beta.1

0b5vr committed:
Merge branch 'dev' into animation-beta

0b5vr committed:
deps (VRMAnimation): Bump Three.js to r160

0b5vr committed:
fix (MToon): Workaround typing of `UniformsUtils.merge`

not a proper typing, might want to fix from three-ts-types side

0b5vr committed:
v2.1.0-beta.2

ke456-png commented on packages/three-vrm-animation/examples/dnd.html:
```diff
@@ -0,0 +1,281 @@
+				"imports": {
+					"three": "https://unpkg.com/three@0.160.0/build/three.module.js",
+					"three/addons/": "https://unpkg.com/three@0.160.0/examples/jsm/",
+					"@pixiv/three-vrm": "https://unpkg.com/@pixiv/three-vrm@2.0.5/lib/three-vrm.module.js",
```
```suggestion
					"@pixiv/three-vrm": "https://unpkg.com/@pixiv/three-vrm@2.0.8/lib/three-vrm.module.js",
```

ke456-png commented on packages/three-vrm-animation/examples/dnd.html:
```diff
@@ -0,0 +1,281 @@
+			const scene = new THREE.Scene();
+
+			// light
+			const light = new THREE.DirectionalLight( 0xffffff );
```
```suggestion
			const light = new THREE.DirectionalLight( 0xffffff, Math.PI );
```

ke456-png commented on packages/three-vrm-animation/examples/loader-plugin.html:
```diff
@@ -0,0 +1,151 @@
+				"imports": {
+					"three": "https://unpkg.com/three@0.160.0/build/three.module.js",
+					"three/addons/": "https://unpkg.com/three@0.160.0/examples/jsm/",
+					"@pixiv/three-vrm": "https://unpkg.com/@pixiv/three-vrm@2.0.5/lib/three-vrm.module.js",
```
```suggestion
					"@pixiv/three-vrm": "https://unpkg.com/@pixiv/three-vrm@2.0.8/lib/three-vrm.module.js",
```

ke456-png commented on packages/three-vrm-animation/examples/loader-plugin.html:
```diff
@@ -0,0 +1,151 @@
+				const scene = new THREE.Scene();
+
+				// light
+				const light = new THREE.DirectionalLight( 0xffffff );
```
```suggestion
				const light = new THREE.DirectionalLight( 0xffffff, Math.PI );
```

ke456-png reviewed (changes requested):
Fixes for r160.

0b5vr committed:
examples: Apply suggestions from code review

- Bump three-vrm to v2.0.8
- Set light intensity to PI

Co-authored-by: ke456-png <108649297+ke456-png@users.noreply.github.com>

0b5vr committed:
Merge branch 'dev' into animation-beta

0b5vr committed:
chore: align package version with 2.0.10 for now

0b5vr committed:
ci: publish workflow, remove dev branch pin

0b5vr committed:
fix: Fix repository field of package.json for vrm animation packages

GitHub provenance failed because of this

0b5vr committed:
fix: Fix parsing animation from VRMA with scales

setFromRotationMatrix don't work with world matrices with scales
replaced with decompose

0b5vr committed:
feature: Promote VRM Animation to the ratified version

Also add a specVersion gate which only passes "1.0" and "1.0-draft"
